### PR TITLE
link/kprobe: Allow to parse symbol offset

### PR DESCRIPTION
**Pull request by nickcooper-zhangtonghao**
Signed-off-by: Tonghao Zhang <zhangtonghao@didiglobal.com>

### CODE REVIEW

File: link/kprobe.go [hunk 1]
```diff
@@ -234,8 +256,12 @@ func pmuProbe(typ probeType, args probeArgs) (*perfEvent, error) {
 		}
 
 		attr = unix.PerfEventAttr{
+			// The minimum size required for PMU uprobes is PERF_ATTR_SIZE_VER1,
```
Comment (mmat11) [inline link/kprobe.go:259]:
```suggestion
			// The minimum size required for PMU kprobes is PERF_ATTR_SIZE_VER1,
```

File: link/kprobe.go [hunk 2]
```diff
@@ -398,6 +427,9 @@ func createTraceFSProbeEvent(typ probeType, args probeArgs) error {
 		//
 		// See Documentation/trace/uprobetracer.txt for more details.
 		pe = fmt.Sprintf("%s:%s/%s %s", probePrefix(args.ret), args.group, args.symbol, uprobeToken(args))
+		if args.offset != 0 {
```
Comment (mmat11) [inline link/kprobe.go:430]: this is not needed, uprobe offset is already part of `uprobeToken`

File: link/kprobe.go [hunk 3]
```diff
@@ -154,10 +171,16 @@ func kprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions, ret bool) (*
 		return nil, fmt.Errorf("eBPF program type %s is not a Kprobe: %w", prog.Type(), errInvalidInput)
 	}
 
+	symbol, offset, err := parseSymbolOffset(symbol)
```
Comment (mmat11) [inline link/kprobe.go:174]: why not just pass the offset via options?

File: link/kprobe.go [hunk 4]
```diff
@@ -386,6 +412,9 @@ func createTraceFSProbeEvent(typ probeType, args probeArgs) error {
 		// the eBPF program itself.
 		// See Documentation/kprobes.txt for more details.
 		pe = fmt.Sprintf("%s:%s/%s %s", probePrefix(args.ret), args.group, args.symbol, args.symbol)
+		if args.offset != 0 {
```
Comment (mmat11) [inline link/kprobe.go:415]: can you put this login in a function similar to `uprobeToken` -> `kprobeToken(symbol, offset)` ?

Comment (nickcooper-zhangtonghao) [reply link/kprobe.go:415]: Thanks! I will send v2

**Pull request closed without merging**